src/Intervals.jl

```diff
@@ -18,23 +19,52 @@ struct Unbounded <: Bound end
 
 bound_type(x::Bool) = x ? Closed : Open
 
-abstract type AbstractInterval{T, L <: Bound, R <: Bound} end
+# Methods for convert between int and tuple representations for space efficiency
+# (Open, Closed) is 16 bytes while the integer represenation is only 1 byte.
+# TODO: Convert types to symbols when we switch to extending IntervalSets.jl
```

Comment:
I've been catching up with the motivation behind these changes. Is there rational why using symbols is a better approach than using types?

Reply:
I don't feel like it matters either way. My preference is more focused on making things consistent between the two, so Intervals.jl can extend IntervalSets.jl. As it stands, there doesn't seem to be much advantage to having them be types. We don't leverage the type hierarchy much and we don't store any information in them, so it's just extra names/types to remember. The practical benefit of switching to symbols is that it means we may be able to have Intervals.jl extend IntervalSets.jl with only one breaking release in Intervals.jl. Going the other way around would require a breaking release to IntervalSets.jl and a lot more updates in the ecosystem.